Userland/Libraries/LibWeb/DOM/Node.cpp

```diff
@@ -2316,9 +2317,11 @@ ErrorOr<String> Node::name_or_description(NameOrDescription target, Document con
             // - Otherwise, return the value of aria-label.
             return element->aria_label().value();
         }
-        // TODO: D. Otherwise, if the current node's native markup provides an attribute (e.g. title) or element (e.g. HTML label) that defines a text alternative,
+        //  E. Otherwise, if the current node's native markup provides an attribute (e.g. title) or element (e.g. HTML label) that defines a text alternative,
         //      return that alternative in the form of a flat string as defined by the host language, unless the element is marked as presentational (role="presentation" or role="none").
-
+        if (is<HTML::HTMLImageElement>(*element))
+            if (auto const& img = static_cast<HTML::HTMLImageElement const&>(*element); img.has_attribute(HTML::AttributeNames::alt))
+                return element->get_attribute(HTML::AttributeNames::alt).value();
```

Comment:
This is only a partial implementation of this spec, can we get a FIXME here?

This seems like a situation where a virtual `alternative_text()` getter on `DOM::Node` would be nice.